mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -437,13 +437,13 @@ impl S3CrtClient {
 /// virtual-hosted-style addresses. The `path_prefix` is appended to the front of all paths, and
 /// need not be terminated with a `/`.
 #[derive(Debug)]
-struct S3Message<'a> {
-    inner: Message<'a>,
+struct S3Message {
```

Comment:
Why was lifetime required earlier and removed now? It might be trivial but I am failing to understand it.

Reply:
That's because it previously owned (through `Message`) an `InputStream<'a>`, which in turn could have its lifetime bound by that of a non-owned buffer.
Now `Message` owns an `AsyncInputStream` which does not have lifetime bounds.